### ISSUE TITLE
trouble converting sql to sqlite

### ISSUE THREAD

**Issue by kelle** [ivh/AstroObjectSchema#9]
I'm not able to successfully convert the sql code generated by the wwwsqldesigner into an sqlite database (I get error messages and an sqlite db that only has one table). what method did you use @Griatch?

**Comment by Griatch** [comment]
The process is documented [here in the wiki](https://github.com/ivh/AstroObjectSchema/wiki/Using-the-XML-file), or is that not working for you?

**Comment by kelle** [comment]
I hadn't found that -- thank you!!!

**Comment by Griatch** [comment]
Updated the page a little: the file examples were a little confusingly named. :)

**Comment by kelle** [comment]
the sqllite database doesn't have the foreign keys defined. Do you know if there's a flag or something which needs to be set in the conversion process from sql to sqlite3?

**Comment by Griatch** [comment]
No, don't think there was anything done in particular to turn foreignkeys on... I suppose the sqlite file in the repo has them, right (I didnt use it myself but I seem to recall you did)? The main thing is that one needs to manually clean some end parens in the dump file as mentione in the wiki.

**Comment by kelle** [comment]
The sqlite in the repo does _not_ have them even though we set them up in the webdesigner.

**Comment by Griatch** [comment]
Hm,  the Mysql file does have the foreign keys, so it does indeed look like something's falling off in the MySQL to SQLite conversion.

**Comment by kelle** [comment]
yep...i see the code in the mysql defining the foreign keys but alas, no foreign keys in the sqlite. We just added them manually in sqlitebrowser and are forging ahead.

**Comment by Griatch** [comment]
Until the convertion proccess can handle it, maybe push your manual .sqlite file to the repo?

**Comment by kelle** [comment]
good idea!

**Comment by Griatch** [comment]
@kelle

I've looked around for various other tools but it appears converting mysql data to sqlite3 is not that straightforward (at least in the open-source world). I've tried a bunch of different conversion scripts with no better luck - the best result so far is mysql2sqlite that we already use. It might be needed to either contribute to [mysql2sqlite](https://github.com/dumblob/mysql2sqlite) and try to figure out why it drops the foreign keys or to make some parser of the wwwsqldesigner's xml file.

**Comment by dr-rodriguez** [comment]
Just to chime in on this: given the way the schema is currently written the foreign keys will not work because in SQLite ALTER TABLE only allows you to rename the table or add a column. If you want to add foreign keys in a way that's understood by both SQLite and MySQL you need to have them explicitly in the CREATE TABLE statements. For example:

```
CREATE TABLE `ObjectReferences` (
  `id` INTEGER NOT NULL,
  `object_id` INTEGER NOT NULL DEFAULT NULL,
  `reference_id` INTEGER NULL DEFAULT NULL,
  `comment` MEDIUMTEXT NULL DEFAULT NULL,
  PRIMARY KEY (`id`),
  FOREIGN KEY (object_id) REFERENCES `Objects` (`id`),
  FOREIGN KEY (reference_id) REFERENCES `Reference` (`id`)
);
```
